packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useCreateFolderPath.ts

```diff
@@ -12,32 +14,29 @@ import {
  * @returns The ID of the final folder in the path.
  */
 export function useCreateFolderPath() {
-  const { synapseClient } = useSynapseContext()
-
   const { mutateAsync: createEntity } = useCreateEntity()
 
+  const queryClient = useQueryClient()
+  const getEntityLookupQueryOptions = useGetEntityLookupQueryOptions()
+  const getEntityQueryOptions = useGetEntityQueryOptions()
+
   return useMutation({
     mutationFn: async (args: { rootContainerId: string; path: string[] }) => {
       const { rootContainerId: rootContainerId, path } = args
       let parentId = rootContainerId
       for (const pathElement of path) {
-        const foundEntityIdResult = await allowNotFoundError(() =>
-          synapseClient.entityServicesClient.postRepoV1EntityChild({
-            entityLookupRequest: {
-              parentId: parentId,
-              entityName: pathElement,
-            },
+        const foundEntityId = await queryClient.fetchQuery(
+          getEntityLookupQueryOptions({
+            parentId: parentId,
+            entityName: pathElement,
           }),
         )
 
-        const foundEntityId = foundEntityIdResult?.id ?? null
-
         if (foundEntityId) {
           // Verify that this entity is a folder
-          const entity =
-            await synapseClient.entityServicesClient.getRepoV1EntityId({
-              id: foundEntityId,
-            })
+          const entity = await queryClient.fetchQuery(
+            getEntityQueryOptions(foundEntityId),
+          )
```

Comment:
We only use this request to see if the entity is a folder or not, so it's safe to cache.